### ISSUE TITLE
[Bug]: Release tarballs are not complete

### ISSUE THREAD

**Issue by didier13150** [netdata/netdata#12169]
### Bug description

Hi there,
Perhaps I misunderstood the process, but downloaded tarballs are not complete and generate compilation errors. 
To workaround this problem, and obtain a successful build, I need to perform additional download and decompress them at their own locations
* netdata/aclk-schemas
* nlohmann/json
* davisking/dlib
* underhood/mqtt_websockets
* underhood/MQTT-C
* underhood/c-rbuf
When I try the master branch via git, all this modules was provided by git submodule, but directories are empty in the released tarballs (version >= 1.33.0)

### Expected behavior

Tarballs with submodule in release, exactly like in git master branch and/or updated compilation doc

### Steps to reproduce

1. Download tarball (>= 1.33.0)
2. Try to compile 
3.
...

### Installation method

from source

### System info

```shell
Linux localhost.localdomain 5.16.7-200.fc35.x86_64 #1 SMP PREEMPT Sun Feb 6 19:53:54 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
/etc/fedora-release:Fedora release 35 (Thirty Five)
/etc/os-release:NAME="Fedora Linux"
/etc/os-release:VERSION="35 (KDE Plasma)"
/etc/os-release:ID=fedora
/etc/os-release:VERSION_ID=35
/etc/os-release:VERSION_CODENAME=""
/etc/os-release:PLATFORM_ID="platform:f35"
/etc/os-release:PRETTY_NAME="Fedora Linux 35 (KDE Plasma)"
/etc/os-release:ANSI_COLOR="0;38;2;60;110;180"
/etc/os-release:LOGO=fedora-logo-icon
/etc/os-release:CPE_NAME="cpe:/o:fedoraproject:fedora:35"
/etc/os-release:REDHAT_BUGZILLA_PRODUCT="Fedora"
/etc/os-release:REDHAT_BUGZILLA_PRODUCT_VERSION=35
/etc/os-release:REDHAT_SUPPORT_PRODUCT="Fedora"
/etc/os-release:REDHAT_SUPPORT_PRODUCT_VERSION=35
/etc/os-release:VARIANT="KDE Plasma"
/etc/os-release:VARIANT_ID=kde
/etc/redhat-release:Fedora release 35 (Thirty Five)
/etc/system-release:Fedora release 35 (Thirty Five)
```

### Netdata build info

```shell
No success build
```

**Comment by Ferroin** [comment]
Can you please confirm whether you downloaded the source tarball provided by GitHub (https://github.com/netdata/netdata/archive/refs/tags/v1.33.1.tar.gz on the latest release) or the distribution tarball we generate and attach to the release (https://github.com/netdata/netdata/releases/download/v1.33.1/netdata-latest.tar.gz on the latest release)?

**Comment by Ferroin** [comment]
To clarify a bit, the source code tarballs that GitHub automatically adds to any release in a repo are actually essentially useless for any project that uses submodules. They’re just a snapshot of the local repo state with the `.git` directory stripped out (IOW, exactly what you see in the ‘Code’ tab on GitHub), without any other special processing done. As a result, these are expected to not contain submodules, and we can’t really do any fancy logic to handle this in our build system because we also lack a `.git` directory to work with.

In contrast, our own distribution tarballs (usually called `netdata-latest.tar.gz` on our recent releases) are put together as part of our build process when preparing a release, and include all the submodules and other special components, minus all the stuff you don’t really need when just building from the tarball (like most of the linter configs or CI infrastructure).

**Comment by didier13150** [comment]
Exactly, I download from https://github.com/netdata/netdata/archive/v1.33.1/netdata-1.33.1.tar.gz, which is equal to https://github.com/netdata/netdata/archive/refs/tags/v1.33.1.tar.gz, but with a different file name. (both md5sum are equal)
I will give a chance to the other link.

**Comment by didier13150** [comment]
It's working as expected after modify the download url. I will always use the released tarballs.
Thank Austin for your help and your precious time.
